packages/schematics/angular/utility/dependency_spec.ts

```diff
@@ -0,0 +1,259 @@
+    ]);
+  });
+
+  it('adds a dependency section if not present', async () => {
+    const tree = new EmptyTree();
+    tree.create('/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      dependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('adds a package to "devDependencies" when "type" is "dev"', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify({
+        dependencies: {},
+        devDependencies: {},
+      }),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0', { type: DependencyType.Dev });
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      dependencies: {},
+      devDependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('adds a package to "peerDependencies" when "type" is "peer"', async () => {
+    const tree = new EmptyTree();
+    tree.create(
+      '/package.json',
+      JSON.stringify({
+        devDependencies: {},
+        peerDependencies: {},
+      }),
+    );
+
+    const rule = addDependency('@angular/core', '^14.0.0', { type: DependencyType.Peer });
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({
+      devDependencies: {},
+      peerDependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('uses specified manifest when provided via "manifestPath" option', async () => {
+    const tree = new EmptyTree();
+    tree.create('/package.json', JSON.stringify({}));
+    tree.create('/abc/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0', {
+      packageJsonPath: '/abc/package.json',
+    });
+
+    await testRule(rule, tree);
+
+    expect(tree.readJson('/package.json')).toEqual({});
+    expect(tree.readJson('/abc/package.json')).toEqual({
+      dependencies: { '@angular/core': '^14.0.0' },
+    });
+  });
+
+  it('schedules a package install task', async () => {
+    const tree = new EmptyTree();
+    tree.create('/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0');
+
+    const tasks = await testRule(rule, tree);
+
+    expect(tasks[0]).toBeInstanceOf(NodePackageInstallTask);
+    expect(tasks.length).toBe(1);
+  });
+
+  it('schedules a package install task with working directory when "manifestPath" is used', async () => {
+    const tree = new EmptyTree();
+    tree.create('/abc/package.json', JSON.stringify({}));
+
+    const rule = addDependency('@angular/core', '^14.0.0', {
+      packageJsonPath: '/abc/package.json',
+    });
+
+    const tasks = await testRule(rule, tree);
+
+    // Should be an array with a single element with working directory set
+    expect(tasks).toEqual([jasmine.objectContaining({ workingDirectory: '/abc' })]);
```

Comment:
Nit: I'm a little worried looking for `workingDirectory` might be too brittle. We could check for `instanceof NodePackageInstallTask` for all of these, but internal changes to that task could still break these tests.

As an alternative, I wonder if we should consider:

```typescript
expect(tasks).toEqual([ new NodePackageInstallTask({ workingDirectory: '/abc' }) ]);
```

Or we could spy on the constructor and return some stubbed object, but my experience with Jasmine has taught me that is more difficult than it should be.

I'm not convinced that's totally better, since it assumes `new NodePackageInstallTask()` is pure / hermetic, so I'll leave that up to you as a suggestion.

Reply:
Switch to testing the output of `toConfiguration` on the tasks array elements which is only an object literal and contains the relevant information the needs to be present for the actual task anyway.  This avoids the entire `instanceof` checking situation.